docs/guide/search/extend_solr.md

```diff
@@ -0,0 +1,542 @@
+    regardless of current user permissions.
+    If you use PHP API in your custom document field mappers, apply [`sudo()`](../../api/public_php_api.md#using-sudo),
+    or use the Persistence SPI layer as in the example above.
+
+## Custom Search Criteria
+
+To provide support for a custom Search Criterion, do the following.
+
+First, in the `src/Query/Criterion/CameraManufacturerVisitor.php` file, implement `CriterionVisitor`:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Query\Criterion;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use EzSystems\EzPlatformSolrSearchEngine\Query\CriterionVisitor;
+
+final class CameraManufacturerVisitor extends CriterionVisitor
+{
+    public function canVisit(Criterion $criterion)
+    {
+        return $criterion instanceof CameraManufacturer;
+    }
+    public function visit(Criterion $criterion, CriterionVisitor $subVisitor = null)
+    {
+        $expressions = array_map(
+            static function ($value): string {
+                return 'exif_camera_manufacturer_id:"' . $this->escapeQuote($value) . '"';
+            },
+            ($criterion->value
+        );
+        return '(' . implode(' OR ', $expressions) . ')';
+    }
+}
+```
+
+Then, create the `src/Query/Criterion/CameraManufacturerCriterion.php` file and add the Criterion class:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Query\Criterion;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Operator;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Operator\Specifications;
+
+final class CameraManufacturer extends Criterion
+{
+    /**
+     * @param string|string[] $value Manufacturer name(s) to be matched.
+     */
+    public function __construct($value)
+    {
+        parent::__construct(null, null, $value);
+    }
+
+    public function getSpecifications(): array
+    {
+        return [
+            new Specifications(
+                Operator::IN,
+                Specifications::FORMAT_ARRAY,
+                Specifications::TYPE_STRING
+            ),
+            new Specifications(
+                Operator::EQ,
+                Specifications::FORMAT_SINGLE,
+                Specifications::TYPE_STRING
+            ),
+        ];
+    }
+}
+```
+
+Finally, register the visitor as a service.
+
+Search Criteria can be valid for both Content and Location search.
+To choose the search type, use either `content` or `location` in the tag:
+
+``` yaml
+services:
+    App\Query\Criterion\CameraManufacturerVisitor:            
+        tags:
+            - { name: ezpublish.search.solr.query.content.criterion_visitor }
+            - { name: ezpublish.search.solr.query.location.criterion_visitor }    
+```
+
+## Custom Sort Clause
+
+To create a custom Sort Clause, do the following.
+
+First, in the `src/Query/SortClause/ScoreVisitor.php` file, implement `SortClauseVisitor`:
```

Comment:
Again, the order should be different. SortClause first, followed by its Visitor.